v3/src/components/tool-shelf/plugins-button.tsx

```diff
@@ -36,7 +36,8 @@ function PluginSelection({ pluginData }: IPluginSelectionProps) {
   function handleClick() {
     documentContent?.applyUndoableAction(
       () => {
-        const url = `${rootPluginUrl}${pluginData.path}`
+        const baseUrl = `${rootPluginUrl}${pluginData.path}`
+        const url = baseUrl.endsWith("/") ? `${baseUrl}index.html` : baseUrl
```

Comment:
Probably worth a comment here. There may be more transformations we want to do to support legacy documents, etc.